### PR TITLE
feat(api): Accepting all types for attributes values

### PR DESCRIPTION
**Pull request by rashidsp**
**Note:** https://github.com/optimizely/ruby-sdk/blob/0e9980a39e81a19cb368c2188b135d7450551516/lib/optimizely/helpers/validator.rb#L43

ATTRIBUTE_VALID_TYPES.include?(attribute_value.class) can't be used as Fixnum and Bignum are deprecated in ruby 2.4 and Rubocop fails: Lint/UnifiedInteger: Use Integer instead of Fixnum.

### CODE REVIEW

**Comment by mikeng13** [comment]
Going to wait on this until we resolve and merge the Python PR, then we should make sure this matches the requirements over there as well @msohailhussain

**Comment by msohailhussain** [comment]
build

File: lib/optimizely/event_builder.rb [hunk 1]
```diff
@@ -74,9 +76,9 @@ def get_common_params(user_id, attributes)
       visitor_attributes = []
 
       attributes&.keys&.each do |attribute_key|
-        # Omit null attribute values
+        # Omit invalid attribute values.
```
Comment (msohailhussain) [inline lib/optimizely/event_builder.rb]: correct comment is "Omit attribute values that are not supported by the log endpoint."
we need to check attribute key is string or not?

File: spec/condition_spec.rb [hunk 2]
```diff
@@ -145,4 +145,19 @@
     condition = JSON.parse(condition)
     expect(@condition_evaluator.evaluate(condition)).to be true
   end
+
+  it 'should evaluate to true for evaluate when user attributes evaluate true' do
```
Comment (mikeng13) [inline spec/condition_spec.rb]: Can you re-word this: `should evaluate to true when user attributes match the audience conditions`

File: spec/event_builder_spec.rb [hunk 3]
```diff
@@ -125,6 +125,70 @@
     expect(impression_event.http_verb).to eq(:post)
   end
 
+  it 'should create a valid Event when create_impression_event is called with attributes of different valid types' do
+    @expected_impression_params[:visitors][0][:attributes].unshift(
```
Comment (mikeng13) [inline spec/event_builder_spec.rb]: Can you copy this into a new variable and mutate that one for use instead?

File: spec/event_builder_spec.rb [hunk 4]
```diff
@@ -125,6 +125,70 @@
...
+      'double_key' => 5.5
+    }
+    impression_event = @event_builder.create_impression_event(experiment, '111128', 'test_user', attributes)
+    expect(impression_event.params).to eq(@expected_impression_params)
+    expect(impression_event.url).to eq(@expected_endpoint)
+    expect(impression_event.http_verb).to eq(:post)
+  end
+
+  it 'should create a valid Event and exclude attributes of invalid types' do
+    @expected_impression_params[:visitors][0][:attributes].unshift(
```
Comment (mikeng13) [inline spec/event_builder_spec.rb]: Same as above

File: lib/optimizely/decision_service.rb [hunk 5]
```diff
@@ -362,17 +362,16 @@ def get_bucketing_id(user_id, attributes)
...
-      bucketing_id = user_id
+      # Returns String representing bucketing ID if it is a String type in attributes else return user ID
 
-      # If the bucketing ID key is defined in attributes, then use that in place of the userID
-      if attributes && attributes[Optimizely::Helpers::Constants::CONTROL_ATTRIBUTES['BUCKETING_ID']].is_a?(String)
-        unless attributes[Optimizely::Helpers::Constants::CONTROL_ATTRIBUTES['BUCKETING_ID']].empty?
-          bucketing_id = attributes[Optimizely::Helpers::Constants::CONTROL_ATTRIBUTES['BUCKETING_ID']]
-          @config.logger.log(Logger::DEBUG, "Setting the bucketing ID '#{bucketing_id}'")
-        end
+      attributes ||= {}
```
Comment (mikeng13) [inline lib/optimizely/decision_service.rb]: Let's return the user_id if attributes is not defined or null

**Comment by mikeng13** [comment]
@rashidsp looks like the test failures are legit. There are some lint errors.

**Pull request merged**